File: two_sum/two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_num = {}

        for i, num in enumerate(nums):
            substract = target - num
            if substract in hash_num:
                return [i, hash_num[substract]]
            hash_num[num] = i
        return []

File: two_sum/two_sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut hash_num: HashMap<i32, i32> = HashMap::new();

        for(i, &num) in nums.iter().enumerate() {
            let substract = target - num;
            if let Some(&index) = hash_num.get(&substract) {
                return vec![i as i32, index as i32];
            }
            hash_num.insert(num, i as i32);
        }

        vec![]
    }
}